### ISSUE TITLE
Error during the transformation

### ISSUE THREAD

**Issue by Thomasbrinn** [dice-project/DICER#1]
Hi, when I start the transformation an internal server error is generated (Error 500). 
In the dicer-service.log file the following error is reported:

Error during module loading: Cannot create a resource for 'c:\Users\Thomas\Desktop\dicer-full/DICER/dicer-core/transformations/uml2tosca.emftvm'; a registered resource factory is needed

How can I fix this error?

**Comment by MicheleGuerriero** [comment]
Hello,

from the error, it looks to me like you are running on Windows.
Unfortunately so far there has been no support for DICER on Windows so I
cannot guarantee it's going to work, but I can try to help you in the case
you want to adjust it to make it run on Windows.
From the error message it looks like it's a path-related issue ("7" vs
"\"). Can you show me how you configuring the config.yml file needed to run
the dicer-service?


Best,

Michele Guerriero

**Comment by Thomasbrinn** [comment]
Hi Michele Guerriero, thank you for your help!
I left the config.yml file unchanged. This is the file:

server:
  type: simple
  applicationContextPath: /
  connector:
    type: http
    port: 8176
transformationDir: ${TRANSFORMATION_DIR}
ddsmMetamodelPath: ${DDSM_METAMODEL_PATH}
diceProfilePath: ${DICE_PROFILE_PATH}
secureUmlProfilePath: ${SECUREUML_PROFILE_PATH}
outMetamodelPath: ${OUT_METAMODEL_PATH}

**Comment by MicheleGuerriero** [comment]
Hello,

did you exported those environment variables (TRANSFORMATION_DIR,
DDSM_METAMODEL_PATH, etc) to point to the right artifacts
(dicer-core/transformations, ddsm.ecore, etc.)?

Michele Guerriero

**Comment by Thomasbrinn** [comment]
Yes, I've done it. I've create the following environment variables:  

TRANSFORMATION_DIR: C:\Users\Thomas\Desktop\dicer-full\DICER\dicer-core\transformations\
DDSM_METAMODEL_PATH: C:\Users\Thomas\Desktop\dicer-full\DICER\dicer-core\metamodels\ddsm.ecore
DICE_PROFILE_PATH: C:\Users\Thomas\Desktop\dicer-full\DICER\dicer-core\metamodels\DICE.profile.uml
SECUREUML_PROFILE_PATH: C:\Users\Thomas\Desktop\dicer-full\DICER\dicer-core\metamodels\SecureUML.profile.uml
OUT_METAMODEL_PATH: C:\Users\Thomas\Desktop\dicer-full\DICER\dicer-core\metamodels\tosca.ecore

**Comment by Thomasbrinn** [comment]
UPDATE:

I've changed the config.yaml file in this way: 

server:
  type: simple
  applicationContextPath: /
  connector:
    type: http
    port: 8176
transformationDir: "C:\\Users\\Thomas\\Desktop\\dicer-full\\DICER\\dicer-core\\transformations"
ddsmMetamodelPath: "C:\\Users\\Thomas\\Desktop\\dicer-full\\DICER\\dicer-core\\metamodels\\ddsm.ecore"
diceProfilePath: "C:\\Users\\Thomas\\Desktop\\dicer-full\\DICER\\dicer-core\\metamodels\\DICE.profile.uml"
secureUmlProfilePath: "C:\\Users\\Thomas\\Desktop\\dicer-full\\DICER\\dicer-core\\metamodels\\SecureUML.profile.uml"
outMetamodelPath: "C:\\Users\\Thomas\\Desktop\\dicer-full\\DICER\\dicer-core\\metamodels\\tosca.ecore"

and now there is no the path-related issue, but I've the same Error, now like this: 

Error during module loading: Cannot create a resource for 'c:\Users\Thomas\Desktop\dicer-full\DICER\dicer-core\transformationsuml2tosca.emftvm'; a registered resource factory is needed

**Comment by MicheleGuerriero** [comment]
Hi think there is still a missing \ at the end of the variables.



Il ven 19 lug 2019, 16:05 Thomasbrinn <notifications@github.com> ha scritto:

**Comment by Thomasbrinn** [comment]
UPDATE:

I've changed the config.yaml file in this way: 

server:
  type: simple
  applicationContextPath: /
  connector:
    type: http
    port: 8176
transformationDir: "C:\\Users\\Thomas\\Desktop\\dicer-full\\DICER\\dicer-core\\transformations\\"
ddsmMetamodelPath: "C:\\Users\\Thomas\\Desktop\\dicer-full\\DICER\\dicer-core\\metamodels\\ddsm.ecore"
diceProfilePath: "C:\\Users\\Thomas\\Desktop\\dicer-full\\DICER\\dicer-core\\metamodels\\DICE.profile.uml"
secureUmlProfilePath: "C:\\Users\\Thomas\\Desktop\\dicer-full\\DICER\\dicer-core\\metamodels\\SecureUML.profile.uml"
outMetamodelPath: "C:\\Users\\Thomas\\Desktop\\dicer-full\\DICER\\dicer-core\\metamodels\\tosca.ecore"

and now there is no the path-related issue, but I've the same Error, now like this: 

Error during module loading: Cannot create a resource for 'c:\Users\Thomas\Desktop\dicer-full\DICER\dicer-core\transformations\uml2tosca.emftvm'; a registered resource factory is needed

**Comment by VALEROPE** [comment]
Hi! I'm the same issue and I'm working on a Mac

**Comment by MicheleGuerriero** [comment]
Hello, could you also post the error you are getting on Mac? Also when does this error occur? When you launch the dicer-service, or when you send it a POST request to transform a model into the TOSCA blueprint?
